content/articles/permission-handling-in-jetpack-compose/index.md

```diff
@@ -0,0 +1,348 @@
+                        it.hasPermission -> {
+                            /* Permission has been granted by the user.
+                               You can use this permission to now acquire the location of the device.
+                               You can perform some other tasks here.
+                            */
+                            Text(text = "Read Ext Storage permission has been granted")
+                        }
+                        it.shouldShowRationale -> {
+                            /*Happens if a user denies the permission two times
+
+                             */
+                            Text(text = "Read Ext Storage permission is needed")
+                        }
+                        !it.hasPermission && !it.shouldShowRationale -> {
+                            /* If the permission is denied and the should not show rationale
+                                You can only allow the permission manually through app settings
+                             */
+                            Text(text = "Navigate to settings and enable the Storage permission")
+
+                        }
+                    }
+                }
+                Manifest.permission.ACCESS_FINE_LOCATION -> {
+                    when {
+                        it.hasPermission -> {
+                            /* Permission has been granted by the user.
+                               You can use this permission to now acquire the location of the device.
+                               You can perform some other tasks here.
+                            */
+                            Text(text = "Location permission has been granted")
+                        }
+                        it.shouldShowRationale -> {
+                            /*Happens if a user denies the permission two times
+
+                             */
+                            Text(text = "Location permission is needed")
+
+                        }
+                        !it.hasPermission && !it.shouldShowRationale -> {
+                            /* If the permission is denied and the should not show rationale
+                                You can only allow the permission manually through app settings
+                             */
+                            Text(text = "Navigate to settings and enable the Location permission")
+
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
+```
+>NOTE: Make sure you annotate these functions with `@Composable`. Also, be sure you call each one of them in the main activity.
+
+```kotlin
+class MainActivity : ComponentActivity() {
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContent {
+            PermissionHandlingComposeDEmoTheme {
+                //SinglePermission()
+                MultiplePermissions()
+            }
+        }
+    }
+}
+```
+
```

Comment:
You can link the GitHub repo with the sample code of this project here. Also, you can have a downloadable sample APK which you can host on Google drive.